M2/Macaulay2/m2/ringmap.m2

```diff
@@ -221,8 +221,9 @@ tensor(RingMap, Matrix) := Matrix => {} >> opts -> (f, m) -> (
     map(f ** target m, f ** source m, f cover m))
 RingMap ** Matrix := Matrix => (f, m) -> tensor(f, m)
 
-VisibleList / RingMap := VisibleList => (v,f) -> apply(v,x -> f x)
-RingMap \ VisibleList := VisibleList => (f,v) -> apply(v,x -> f x)
+VisibleList / RingMap     := VisibleList => (v,f) -> apply(v,x -> f x)
+RingMap     \ VisibleList :=
+RingElement \ VisibleList := VisibleList => (f,v) -> apply(v,x -> f x)
```

Comment:
This _looks_ like two similar methods, but really they are completely different:
1. `RingMap \ List` evaluates a single ring map with multiple inputs
2. `RingElement \ List` creates multiple ring maps and evaluates each with the same input.

Besides this conflation, on the surface this is a convenient improvement, but underneath this is bad for efficiency because creating throwaway new ring maps using `substitute` is slow and shouldn't be made more convenient.

I'm not in favor of this change.

Reply:
Ah, good point.  Are you suggesting that we scrap this idea entirely, or just re-implement by creating a single ring map?

Reply:
I don't know how to do this with a single ring map, but `(x -> x^2) \ {1, 2, 3}` is a simple enough workaround that I would close this PR.

To be honest, I've found #2863 convenient, but also found it to be a surprising bottleneck several times and regretted it. It's plausible that we can come back to this PR when `sub` is significantly more efficient though, along with other useful features like #2882.

Reply:
Fair enough -- I'll close this